crates/hyperdrive-math/src/short/open.rs

```diff
@@ -50,11 +49,15 @@ impl State {
         )
     }
 
-    /// Gets the spot price after opening the short on the YieldSpace curve and
-    /// before calculating the fees.
-    pub fn get_spot_price_after_short(&self, bond_amount: FixedPoint) -> FixedPoint {
-        let shares_amount = self.calculate_shares_out_given_bonds_in_down(bond_amount);
-        self.spot_price_after_short(shares_amount * self.vault_share_price(), bond_amount)
+    /// Gets the spot price after opening a Hyperdrive short.
+    pub fn get_spot_price_after_short(
+        &self,
+        short_amount: FixedPoint,
+        spot_price: FixedPoint,
+        open_vault_share_price: FixedPoint,
+    ) -> Result<FixedPoint> {
+        self.calculate_open_short(short_amount, spot_price, open_vault_share_price)
```

Comment:
If we're using this function to check for negative interest, do we not want to call the `calculate_open_short` function here? This would result in infinite recursion.

Reply:
This is the `pub` function `get_spot_price_after_short` which calls `calculate_open_short` to determine the `base_amount`/`short_deposit`.

`calculate_open_short` uses the internal `spot_price_after_short` which takes a specified `base_amount`.

However, @dpaiton has a PR in which these can be consolidated and `calculate_open_short` will only be called if no `base_amount` id provided that I'll probably copy here.